### PR TITLE
Add single quote to `iskeyword`

### PR DESCRIPTION
**Pull request by Aster89**
The single quote `'` is a valid character in variable names, so it should be included in `iskeyword`; this change, for instance, makes the <kbd>*</kbd> command behave predictably, which is good.

### CODE REVIEW

**Comment by brammool** [comment]
Did you contact the maintainer?  See the file header.

**Comment by Aster89** [comment]
I'm sorry, @brammool , I hadn't read the [CONTRIBUTING.md](https://github.com/vim/vim/blob/master/CONTRIBUTING.md) file. I've just sent an e-mail to him now.

**Comment by marcotrosi** [comment]
you can change _iskeyword_ anytime or set it in your vimrc, why making the single quote default? I don't think it's an allowed character in every language, is it?

**Comment by k-takata** [comment]
This is a change for Haskell.

**Comment by marcotrosi** [comment]
> This is a change for Haskell.

oopsie, I obviously missed that important info. Thx for the clarification.

**Comment by sol** [comment]
1. While this works in some trivial cases, it breaks *-search for any practical purposes. [^1]
1. I'm not sure if it's a good idea to change Haskell related stuff without having anybody who has significant experience with Haskell being involved.
1. May I politely request to revert this change?

[^1]: It doesn't work with `.chs` files e.g. consider something like \`Char', it doesn't play nice with Haddock comments, e.g. consider `use 'someFunction' to...`, it doesn't play nice with promotion and TH quotes e.g. consider `'Foo` or `'foo`, and probably more…

**Comment by chrisbra** [comment]
sorry, I don't understand your footnotes. Can you please clarify what exactly breaks? Also it looks like I forgot to update `b:undo_ftplugin`, so it might be a good idea to revert this indeed.

**Comment by sol** [comment]
```haskell
-- | This is the documentation of function 'foo'.
foo = 'bar

bar :: 'True
bar = undefined
```
- `'foo'` in the comment refers to `foo`.
- `'bar` in the definition of `foo` refers to `bar` (or rather the TH `Name` of `bar` aka quoted).
- `'True` refers to `True` promoted to the type level.
 
 If you *-search any of `foo`, `bar` or `True` then you want a match for all occurrences, including quotations, promotions, and in documentation.

(I'm skipping the discussion of `.chs` files here)

There are also function names that contain a `'`, e.g. `foldl` and `foldl'` refer to different functions. So you can make a case for adding `'` to `iskeyword`.  But if you do that you will eventually come to the realization that this is not practical, that the downsides far outweigh the benefits.

The bottom line is this:
- Having *-search match both `foldl` and `foldl'` is a minor inconvenience at worst.
- On the other hand, having *-search miss valid matches is a major pain.

I can't speak for others, but what I did 15 years ago is added `'` to `iskeyword`, realized that this just does not work and removed it again.  I assume that's the journey most Haskell developers go through.

**Comment by chrisbra** [comment]
okay, reverted as of bda55df3b

Thanks.

**Comment by Aster89** [comment]
Apologies for the change in the first place. I didn't know there were so many cases that would be broken by this change :disappointed:
